src/poker/game/GamePlay.java

```diff
@@ -0,0 +1,362 @@
+package poker.game;
+
+import poker.*;
+
+import java.util.*;
+
+public class GamePlay {
+    public static final Map<GamePhase, Integer> BOARD_PRINT_INDEXES = Map.of(
+            GamePhase.PREFLOP, 3,
+            GamePhase.FLOP, 4,
+            GamePhase.TURN, 5
+    );
+
+    public Deck deck;
+    public HandsAndBoard result;
+    public Player player1;
+    public Player player2;
+    GameState gameState;
+    List<Player> playerList;
+    public EquityEvaluator equityEvaluator;
+    boolean firstDecisionSmallBlind = true;
+    boolean firstDecisionPostFlop = true;
+    boolean playerFold = false;
+    public GameResult gameResult;
+
+
+    public GamePlay() {
+        this.deck = new Deck();
+        this.result = deck.shuffleDeckAndGetHandsAndBoard();
+        this.player1 = new Player(100, 0, 0, 0, result.getHand1(), "player1");
+        this.player2 = new Player(100, 0, 0, 0, result.getHand2(), "player2");
+        this.gameState = new GameState(GamePhase.PREFLOP, 0);
+        player1.setHandPlayer(result.getHand1());
+        player2.setHandPlayer(result.getHand2());
+        playerList = new ArrayList<>();
+        playerList.add(player1);
+        playerList.add(player2);
+        this.equityEvaluator = new EquityEvaluator(deck, result.getHand1(), result.getHand2(), result.getBoard());
+        this.gameResult = new GameResult();
+    }
+
+    public void gameStart() {
+        Player nonActivePlayer = player1;
+        Player activePlayer = player2;
+        Player smallBlindPlayer = activePlayer;
+        Player bigBlindPlayer = nonActivePlayer;
+        boolean OutOfPostionPlayer = false;
+        while (playerList.size() > 1) {
+
+            topUpPlayer(nonActivePlayer, activePlayer);
+            preFlopDetails(nonActivePlayer, activePlayer);
+            PossibleAction lastAction = null;
+            for (int i = 0; i < 4; i++) {
+
+                while (nextTurnPlayer(lastAction, OutOfPostionPlayer) && !playerFold) {
+                    if (gameState.gamePhase != GamePhase.PREFLOP && smallBlindPlayer == activePlayer && firstDecisionPostFlop) {
+                        Player temp = activePlayer;
+                        activePlayer = nonActivePlayer;
+                        nonActivePlayer = temp;
+                    }
+                    if (activePlayer != smallBlindPlayer && gameState.gamePhase != GamePhase.PREFLOP) {
+                        OutOfPostionPlayer = true;
+                    } else {
+                        OutOfPostionPlayer = false;
+                    }
```

Comment:
jak ci inteliJ cos podpowiada to uzyj
![image](https://github.com/user-attachments/assets/5fbad35b-3b47-4e38-9224-dfd0f6e9c784)

Reply:
to zlote podkreslenie